test/socket.io.ts

```diff
@@ -952,8 +955,7 @@ describe("socket.io", () => {
       const sio = new Server(srv);
       srv.listen(() => {
         const clientSocket = client(srv, { reconnection: false });
-        clientSocket.on("connect", function init() {
-          clientSocket.removeListener("connect", init);
+        clientSocket.once("connect", () => {
```

Comment:
Improving the typing of client `Socket`s showed that `clientSocket.removeListener` does not exist in [component-emitter](https://www.npmjs.com/package/component-emitter), so this had to be fixed in order to get tests to compile.